### DISCUSSION TITLE
Sandbox is giving errors but not way to know which is the problem

### DISCUSSION THREAD

**Discussion by ignaciolarranaga** [mercadopago/sdk-react#68] [Q&A] [answered]
Hey, sorry if too naive question, but since today the sandbox started giving me errors when trying to complete a successful payment and I can not find a way to understand what am I doing wrong?
Is there some error log or something?

The only thing I can see is:
<img width="894" alt="Screenshot 2023-07-28 at 08 51 01" src="https://github.com/mercadopago/sdk-react/assets/15932965/d788691d-e4b2-40c7-8c2e-34b8023ede9d">

I tried inspecting the network calls during the payment without success.
I tried it multiple times with both test cards also without success.

If I try to get back to my site the URL looks like the following: `http://localhost:3000/es/compra/i2Vu6rE0Ap3S2BshPAxm/fallida?collection_id=null&collection_status=null&payment_id=null&status=null&external_reference=64c3a68cf42efd623f231904&payment_type=null&merchant_order_id=null&preference_id=8640843-8614ea84-e48f-4de1-81cf-c096ffd5a8f3&site_id=MLU&processing_mode=aggregator&merchant_account_id=null`

It is my backurl format, but can not get any clue from the parameters on what happened.

**Comment by icaldana** [comment]
Hello @ignaciolarranaga!

There are not silly questions, don’t worry 😄 
About the problem, it seems to be related to Checkout Pro payments - I mean you are paying with Wallet/Mercado Pago. I've redirected this issue to the responsible team, so I hope I will have updates soon.

Meanwhile, could you provide more information about the error and what type of payment are you performing?

Thank you.

**Reply by ignaciolarranaga** [reply, marked as answer]
Hey got solved, not sure how, or what am I now doing differently.

I suspected that it might be because of the date and the mask (i.e. typing 1125 instead of 11/25, I mean literally typing the "/", but just double checked and it wasn't the case). May be it was a glitch, who knows 😄 

While testing I created a cypress command, sharing in case useful:
```
export function addPayWithMercadoPagoCommand() {
  Cypress.Commands.add(
    'payWithMercadoPago',
    (
      // spellchecker: disable
      outcome:
        | 'APRO'
        | 'CONT'
        | 'OTHE'
        | 'CALL'
        | 'FUND'
        | 'SECU'
        | 'EXPI'
        | 'FORM'
      // spellchecker: enable
    ) => {
      // Paying with MP
      cy.step('Paying with MP');
      cy.get('div#walletBrick_container button', {
        timeout: 10000,
      }).click();

      cy.origin('https://www.mercadopago.com.uy', () => {
        Cypress.require('cypress-iframe');

        // spellchecker: disable
        // Checking reach the page
        cy.get('h1.c-title', { timeout: 10000 }).should(
          'have.text',
          '¿Cómo querés pagar?'
        );

        // Selecting New Card
        cy.get('label[for=new_card_row]').click();

        // Checking reach the page
        cy.get('h1.c-title', { timeout: 10000 }).should(
          'have.text',
          'Completá los datos de tu tarjeta'
        );

        // Filling the data
        cy.iframe('iframe[name=cardNumber]')
          .find('input#cardNumber')
          .should('exist')
          .type('5031 7557 3453 0604');

        cy.iframe('iframe[name=expirationDate]')
          .find('input#expirationDate')
          .should('exist')
          .type('11/25');

        cy.get(
          'input[name="[ui_components][group_content][group_scroller][group_form][controls][bottom_card_group][fullname]"]'
        ).type(outcome);

        cy.iframe('iframe[name=securityCode]')
          .find('input#securityCode')
          .should('exist')
          .type('123');

        // Submitting
        cy.get(
          'button[name="[ui_components][group_content][group_scroller][group_form][submit]"]'
        ).click();

        // Checking Transition
        cy.get('h1.c-title', { timeout: 10000 }).should(
          'have.text',
          'Ahora ingresá el documento del titular'
        );

        // Filling the data
        cy.get('button.andes-dropdown__trigger').click();
        cy.get('li[data-key="[Otro]"]').should('be.visible').click();

        // Filling the data
        cy.get('input#number').type('123456789');

        // Submitting
        cy.get(
          'button[name="[ui_components][group_content][group_form][submit]"]'
        ).click();

        // Checking Transition
        cy.get('h1.c-title', { timeout: 10000 }).should(
          'have.text',
          'Revisa tu compra'
        );

        // Submitting
        cy.get(
          'button[name="[ui_components][group_sidebar][summary_group][group_captcha][group_pay][disabled_button][pay]"]'
        ).click();

        cy.contains('a', 'Volver al sitio', { timeout: 10000 }).click();
      });
      // spellchecker: enable
    }
  );
}
```

By the way, it would be nice if you guys can add some meaningful IDs to the elements (e.g. data-testid="select-new-card-payment"), it would help to develop automated tests. Would it be even better a cypress command being distributed and supported, hehe

Have a nice weekend and thanks for the help!
P.S.: Introducing the CI does not work in the form, it says invalid CI, you must use Otro. Commenting because of the doc: https://www.mercadopago.com.uy/developers/es/docs/checkout-api/additional-content/your-integrations/test/cards

**Comment by ignaciolarranaga** [comment]
@icaldana it is happening again. Not accepting the testing card. Did you guys discover something the last time?

**Comment by ignaciolarranaga** [comment]
@icaldana reproduction details in this repo and a recording of the issue: https://github.com/ignaciolarranaga/mp-issue

Let me know if useful for diagnosing or you need something else.

**Reply by ignaciolarranaga** [reply]
Hey @icaldana , please dismiss, it was my bad, I was passing an incorrect URL (I missed ":" in the url, http//example.com), once I fixed that everything flew smoothly!, probably the same was happening the other day.

May be nice some log or error message somewhere so we do not have to disturb you guys.

Thanks again!
